### PR TITLE
Clarify instance group documentation for ensemble

### PR DESCRIPTION
**Pull request by Tabrizian**

### CODE REVIEW

File: docs/user_guide/architecture.md [hunk 1]
```diff
@@ -796,6 +796,14 @@ scheduler will:
    response to the inference request with the tensors mapped to the ensemble
    output names.
 
+Unlike other models, ensemble models do not support "instance_group" field in
+the model configuration. The reason is that the ensemble scheduler itself
+is mainly an event-driven scheduler with very minimal overhead so its
+almost never the bottleneck of the pipeline. For optimized throughput,
+make sure to optimize your pipeline using
+[Model Analyzer](https://github.com/triton-inference-server/model_analyzer)
+for best performance.
```
Comment (rmccorm4) [inline docs/user_guide/architecture.md:805]: Re-wording suggestion
```suggestion
almost never the bottleneck of the pipeline. The composing models
within the ensemble can be individually scaled up or down with their
respective `instance_group` settings. To optimize your model pipeline
performance, you can use
[Model Analyzer](https://github.com/triton-inference-server/model_analyzer)
to find the optimal model configurations.
```

Comment (Tabrizian) [reply docs/user_guide/architecture.md:805]: thanks for the suggestion. Fixed.

Comment (rmccorm4) [reply docs/user_guide/architecture.md:805]: I'm not seeing the update on my end, do you?

Comment (Tabrizian) [reply docs/user_guide/architecture.md]: Oops, looks like push failed. Should be available now.

**Review by rmccorm4 — APPROVED** [review]
:rocket:

**Pull request merged**